A device 1.1/2.0 api migration broke some tests

PR https://github.com/tenstorrent/tt-metal/pull/45761 caused a failure:
https://github.com/tenstorrent/tt-metal/actions/runs/27187949553/job/80262093783
```
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_transpose.py::test_transpose_hc_sharded_with_program_cache[20-4-224-224-grid_size5] - RuntimeError: TT_THROW @ /project/tt_metal/impl/program/program.cpp:255: tt::exception
info:
Failed to generate binaries for reader_unary_transpose_hc_sharded_rm TT_THROW @ /project/tt_metal/jit_build/build.cpp:67: tt::exception
info:
ncrisc build failed. Log: In file included from ../kernel_includes.hpp:1,
                 from /opt/venv/lib/python3.10/site-packages/ttnn/tt_metal/hw/firmware/src/tt-1xx/ncrisck.cc:22:
/work/ttnn/cpp/ttnn/operations/data_movement/transpose/device/kernels/dataflow/reader_unary_transpose_hc_sharded_rm.cpp: In function 'void kernel_main()':
/work/ttnn/cpp/ttnn/operations/data_movement/transpose/device/kernels/dataflow/reader_unary_transpose_hc_sharded_rm.cpp:151:9: error: 'noc' was not declared in this scope; did you mean 'Noc'?
  151 |         noc.async_read(
      |         ^~~
      |         Noc
/work/ttnn/cpp/ttnn/operations/data_movement/transpose/device/kernels/dataflow/reader_unary_transpose_hc_sharded_rm.cpp:175:5: error: 'noc' was not declared in this scope; did you mean 'Noc'?
  175 |     noc.async_read_barrier();
      |     ^~~
      |     Noc
```

Needs to be fixed

--- a/ttnn/cpp/ttnn/operations/data_movement/transpose/device/kernels/dataflow/reader_unary_transpose_hc_sharded_rm.cpp
+++ b/ttnn/cpp/ttnn/operations/data_movement/transpose/device/kernels/dataflow/reader_unary_transpose_hc_sharded_rm.cpp
@@ -116,6 +116,7 @@ void kernel_main() {
 
     const uint32_t stick_size_bytes = W_size_bytes;
 
+    Noc noc;
     CircularBuffer cb_in(cb_in0);
     CircularBuffer cb_out(cb_out0);
 

